### PR TITLE
Sync with analysis_options.yaml in flutter/packages

### PR DESCRIPTION
**Pull request by JSUYA**
Some packages require package upgrade rather than simple modification (ex: google_maps_flutter)
Therefore, the rules below are temporarily ignored.
Any necessary modifications can be applied in a later PR.

https://github.com/flutter/packages/pull/4060
- unnecessary_null_comparison

https://github.com/flutter/packages/pull/4067
- unawaited_futures

https://github.com/flutter/packages/pull/5717
- dangling_library_doc_comments
- no_literal_bool_comparisons
- unnecessary_library_directive
- use_colored_box
- use_enums
- use_string_in_part_of_directives

### CODE REVIEW

File: packages/messageport/example/integration_test/messageport_test.dart [hunk 1]
```diff
@@ -26,7 +26,7 @@ void main() {
 
   testWidgets('Create remote port', (WidgetTester tester) async {
     final LocalPort localPort = await LocalPort.create(kTestPort);
-    localPort.register((dynamic message, [RemotePort? remotePort]) => null);
+    localPort.register((dynamic message, [RemotePort? remotePort]) => () {});
```
Comment (swift-kim) [inline packages/messageport/example/integration_test/messageport_test.dart:29]:
```suggestion
    localPort.register((dynamic message, [RemotePort? remotePort]) {});
```

File: analysis_options.yaml [hunk 2]
```diff
@@ -1,16 +1,22 @@
-# This file is a copy of analysis_options.yaml from flutter/packages repo
-# as of 2023-03-07.
+# Specify analysis options.
+#
+# This file is a copy of analysis_options.yaml from flutter repo
+# as of 2023-12-18, but with some modifications marked with
+# "DIFFERENT FROM FLUTTER/FLUTTER" below. The file is expected to
+# be kept in sync with the master file from the flutter repo.
```
Comment (swift-kim) [inline analysis_options.yaml:6]: You may revert this change and just replace `2023-03-07` with `2024-03-07`.

Comment (JSUYA) [reply analysis_options.yaml]: Thank you for your review. I modified it.

File: packages/messageport/example/integration_test/messageport_test.dart (hunk 1 above)
Comment (JSUYA) [reply packages/messageport/example/integration_test/messageport_test.dart]: done

**Review by swift-kim — APPROVED** [review]
